akka-http/src/main/scala/akka/http/scaladsl/marshalling/Marshal.scala

```diff
@@ -16,6 +16,18 @@ object Marshal {
 
   final case class UnacceptableResponseContentTypeException(supported: Set[ContentNegotiator.Alternative])
     extends RuntimeException with NoStackTrace
+
+  private[marshalling] def selectMarshallingForContentType[T](marshallings: Seq[Marshalling[T]], contentType: ContentType): Option[() ⇒ T] = {
+    contentType match {
+      case best @ (_: ContentType.Binary | _: ContentType.WithFixedCharset | _: ContentType.WithMissingCharset) ⇒
```

Comment:
`best` is a bit of a misnomer here, anyway. Could get rid of it and simply have:
```
  contentType match {
    case _: ContentType.Binary | _: ContentType.WithFixedCharset | _: ContentType.WithMissingCharset ⇒
      marshallings collectFirst { case Marshalling.WithFixedContentType(`contentType`, marshal) ⇒ marshal }
    case ContentType.WithCharset(mediaType, charset) ⇒
      marshallings collectFirst {
        case Marshalling.WithFixedContentType(`contentType`, marshal) ⇒ marshal
        case Marshalling.WithOpenCharset(`mediaType`, marshal)    ⇒ () ⇒ marshal(charset)
      }
  }
```